common/src/api/external/mod.rs

```diff
@@ -288,6 +288,18 @@ impl TryFrom<String> for NameOrId {
     }
 }
 
+impl From<Name> for NameOrId {
+    fn from(name: Name) -> Self {
+        NameOrId::Name(name)
+    }
+}
+
+impl From<Uuid> for NameOrId {
+    fn from(id: Uuid) -> Self {
+        NameOrId::Id(id)
+    }
+}
```

Comment:
These greatly reduce a lot of conversion noise. We might not need them post v1 migration.